frame/democracy/src/tests/public_proposals.rs

```diff
@@ -60,16 +60,30 @@ fn deposit_for_proposals_should_be_returned() {
 		assert_ok!(Democracy::second(RuntimeOrigin::signed(5), 0));
 		assert_ok!(Democracy::second(RuntimeOrigin::signed(5), 0));
 		fast_forward_to(3);
-		assert_eq!(Balances::free_balance(1), 10);
+		assert_eq!(Balances::free_balance(1), 0);
 		assert_eq!(Balances::free_balance(2), 20);
 		assert_eq!(Balances::free_balance(5), 50);
 	});
 }
 
 #[test]
-fn proposal_with_deposit_below_minimum_should_not_work() {
+fn proposal_with_funds_below_minimum_should_not_work() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(propose_set_balance(1, 2, 5));
+
+		assert_noop!(propose_set_balance(1, 2, 0), Error::<Test>::InsufficientFunds);
+	});
+}
+
+#[test]
+fn creating_proposal_takes_fee() {
 	new_test_ext().execute_with(|| {
-		assert_noop!(propose_set_balance(1, 2, 0), Error::<Test>::ValueLow);
+		System::set_block_number(0);
+		assert_eq!(Balances::free_balance(1), 10);
+
+		assert_ok!(propose_set_balance(1, 2, 5));
+
+		assert_eq!(Balances::free_balance(1), 0);
```

Comment:
`free_balance` could be reduced by `reserve` or `lock`. We want to actually take the coins, so we should check all balance.